File: src/tools/registry.ts
import type { Tool } from "./types.js";
import { readFileTool } from "./readFile.js";
import { writeFileTool } from "./writeFile.js";
import { editFileTool } from "./editFile.js";
import { listDirTool } from "./listDir.js";
import { globTool } from "./glob.js";
import { grepTool } from "./grep.js";
import { bashTool } from "./bash.js";

export const allTools: Tool[] = [
  readFileTool,
  writeFileTool,
  editFileTool,
  listDirTool,
  globTool,
  grepTool,
  bashTool,
];

export const toolsByName = new Map(allTools.map((t) => [t.definition.name, t]));

File: desktop/core/src/tools/registry.rs
use super::{BashTool, EditFileTool, GlobTool, GrepTool, ListDirTool, ReadFileTool, Tool, WriteFileTool};
use std::collections::HashMap;
use std::sync::Arc;

pub fn all_tools() -> Vec<Arc<dyn Tool>> {
    vec![
        Arc::new(ReadFileTool),
        Arc::new(WriteFileTool),
        Arc::new(EditFileTool),
        Arc::new(ListDirTool),
        Arc::new(GlobTool),
        Arc::new(GrepTool),
        Arc::new(BashTool),
    ]
}

pub fn tools_by_name() -> HashMap<String, Arc<dyn Tool>> {
    all_tools().into_iter().map(|t| (t.definition().name.clone(), t)).collect()
}
